tests/FSharp.AWS.DynamoDB.Tests/UpdateExpressionTests.fs

```diff
@@ -243,6 +243,35 @@ type ``Update Expression Tests``(fixture: TableFixture) =
         let item' = table.UpdateItem(key, <@ fun (r: R) -> { r with String = defaultArg r.Optional "<undefined>" } @>)
         test <@ "<undefined>" = item'.String @>
 
+    [<Fact>]
+    let ``Update using defaultArg combinator on the same optional attribute (None)`` () =
+        let item = { mkItem () with Optional = None }
+        let key = table.PutItem item
+        let item' = table.UpdateItem(key, <@ fun (r: R) -> { r with Optional = defaultArg r.Optional "<undefined>" |> Some } @>)
+        test <@ Some "<undefined>" = item'.Optional @>
+
+    [<Fact>]
+    let ``Update using defaultArg combinator on the same optional attribute (Some)`` () =
+        let item = { mkItem () with Optional = Some(guid ()) }
+        let key = table.PutItem item
+        let item' = table.UpdateItem(key, <@ fun (r: R) -> { r with Optional = defaultArg r.Optional "<undefined>" |> Some } @>)
+        test <@ item.Optional = item'.Optional @>
+
+    [<Fact>]
+    let ``Update using defaultArg combinator on the same non-optional attribute (None)`` () =
+        let item = mkItem ()
+        let key = table.PutItem item
+        clearAttribute table key "String"
+        let item' = table.UpdateItem(key, <@ fun (r: R) -> { r with String = defaultArg (Some r.String) "<undefined>" } @>)
```

Comment:
was about to comment that this is the same as L272 as the `clearAttribute`did not jump out at me as being the important bit
is there any way that 263 could do `let key = table.PutItem { item with .... }` or is it important tha the clear is a separate call?

Reply:
It’s a non-optional field, it can’t be set to `None` on the record. I’m not convinced we need if-not-exists support for these non-optionals, it’s only likely to come about as the result of schema changes, but supporting it shouldn’t break anything important

Reply:
tbh I have very little idea what the behavior and/or schema is - from my perspective it's a small piece of expression handling with supporting tests that will be used so it meets the bar regardless of whether I even get how it works ;)